mmv1/products/firestore/BackupSchedule.yaml

```diff
@@ -0,0 +1,98 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Resource
+name: 'BackupSchedule'
+base_url: projects/{{project}}/databases/{{database}}/backupSchedules
+self_link: '{{name}}'
+immutable: true
+description: |
+  A backup schedule for a Cloud Firestore Database.
+  This resource is owned by the database it is backing up, and is deleted along with the database.
+  The actual backups are not though.
+references: !ruby/object:Api::Resource::ReferenceLinks
+  guides:
+    'Official Documentation': 'https://cloud.google.com/firestore/docs/backups'
+  api: 'https://cloud.google.com/firestore/docs/reference/rest/v1/projects.databases.backupSchedules'
+skip_sweeper: true
+id_format: 'projects/{{project}}/databases/{{database}}/backupSchedules/{{name}}'
+import_format:
+  - 'projects/{{project}}/databases/{{database}}/backupSchedules/{{name}}'
+  - '{{project}}/{{database}}/{{name}}'
+  - '{{database}}/{{name}}'
+  - '{{name}}'
+docs: !ruby/object:Provider::Terraform::Docs
+  warning: |
+    This resource creates a Firestore Backup Schedule on a project that already has
+    a Firestore database. If you haven't already created it, you may
+    create a `google_firestore_database` resource with `type` set to
+    `"FIRESTORE_NATIVE"` and `location_id` set to your chosen location.
+    If you wish to use App Engine, you may instead create a
+    `google_app_engine_application` resource with `database_type` set to
+    `"CLOUD_FIRESTORE"`. Your Firestore location will be the same as
+    the App Engine location specified.
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    name: 'firestore_backup_schedule_basic'
+    primary_resource_id:
+      'daily-backup'
+    test_env_vars:
+      project_id: :FIRESTORE_PROJECT_NAME
+properties:
+  - !ruby/object:Api::Type::String
+    name: name
+    output: true
+    description: |
+      A server defined name for this index. Format:
+      `projects/{{project}}/databases/{{database}}/backupSchedules/{{backupSchedule}}
+  - !ruby/object:Api::Type::String
+    name: database
+    default_value: '(default)'
+    description: |
+      The Firestore database id. Defaults to `"(default)"`.
+    ignore_read: true
+  - !ruby/object:Api::Type::String
+    name: retention
+    required: true
+    description: |
+      At what relative time in the future, compared to its creation time, the backup should be deleted, e.g. keep backups for 7 days.
+      A duration in seconds with up to nine fractional digits, ending with 's'. Example: "3.5s".
```

Comment:
I think this field also needs to be removed from the PR? I can't see the `database` field in [the documentation for a backup schedule](https://cloud.google.com/firestore/docs/reference/rest/v1/projects.databases.backupSchedules), and currently when I try to run acceptance tests I get this error from the API:


```
  |       {
  |         "description": "Invalid JSON payload received. Unknown name \"database\" at 'backup_schedule': Cannot find field.",
  |         "field": "backup_schedule"
  |       }
```

Reply:
Ah I see this is needed to form the URL for the resource - I think that you need to include database as a parameter in the YAML, [similar to this example](https://github.com/GoogleCloudPlatform/magic-modules/blob/c8478f3a43b0f3f03542f5427104760eb5fb482c/mmv1/products/apigateway/ApiConfig.yaml#L93-L107)

Reply:
Ahh yes thanks, I've updated it.
When looking at various resources I've seen both approaches where it's in the parameters and where it's in the properties, that confused me!
e.g. in the Firestore `Field` is was in the properties :)